yt/geometry/oct_geometry_handler.py

```diff
@@ -33,3 +36,83 @@ def get_smallest_dx(self):
 
     def convert(self, unit):
         return self.dataset.conversion_factors[unit]
+
+    def _add_mesh_sampling_particle_field(self, deposit_field, ftype, ptype):
+        units = self.ds.field_info[ftype, deposit_field].units
+        take_log = self.ds.field_info[ftype, deposit_field].take_log
+        field_name = "cell_%s_%s" % (ftype, deposit_field)
+
+        def _cell_index(field, data):
+            # Get the position of the particles
+            pos = data[ptype, "particle_position"]
+            Npart = pos.shape[0]
+            ret = np.zeros(Npart)
+            tmp = np.zeros(Npart)
+
+            if isinstance(data, FieldDetector):
+                return ret
+
+            remaining = np.ones(Npart, dtype=bool)
+            Nremaining = Npart
+
+            Nobjs = len(data._current_chunk.objs)
+            Nbits = int(np.ceil(np.log2(Nobjs)))
+
+            for i, obj in enumerate(data._current_chunk.objs):
+                if Nremaining == 0:
+                    continue
+                icell = obj['index', 'ones'].T.reshape(-1).astype(np.int64).cumsum().value - 1
+                mesh_data = ((icell << Nbits) + i).astype(np.float64)
+                # Access the mesh data and attach them to their particles
+                tmp[:Nremaining] = obj.mesh_sampling_particle_field(pos[remaining], mesh_data)
+
+                ret[remaining] = tmp[:Nremaining]
+
+                remaining[remaining] = np.isnan(tmp[:Nremaining])
+                Nremaining = remaining.sum()
+
+            return data.ds.arr(ret.astype(np.float64), input_units='1')
+
+        def _mesh_sampling_particle_field(field, data):
+            """
+            Create a grid field for particle quantities using given method.
+            """
+            ones = data[ptype, 'particle_ones']
+
+            # Access "cell_index" field
+            cell_data = data[ftype, deposit_field].reshape(-1)
+            Npart = ones.shape[0]
+            ret = np.zeros(Npart)
+            cell_index = np.array(data[ptype, 'cell_index'], np.int64)
+
+            if isinstance(data, FieldDetector):
+                return ret
+
+            # The index of the obj is stored on the first bits
+            Nobjs = len(data._current_chunk.objs)
+            Nbits = int(np.ceil(np.log2(Nobjs)))
+            icell = cell_index >> Nbits
+            iobj = cell_index - (icell << Nbits)
+            for i, subset in enumerate(data._current_chunk.objs):
+                mask = (iobj == i)
+
+                cell_data = subset[ftype, deposit_field].T.reshape(-1)
```

Comment:
This line will return something unexpected for example if `deposit_field` is `('index', 'radius')`. The values returned by subset is given in absolute coordinates while it is expected to be in relative coordinates.